toolz/itertoolz.py

```diff
@@ -97,6 +98,16 @@ def groupby(key, seq):
     return rv
 
 
+def indices(*sizes):
+    """ Iterates over a length/shape.
+
+    >>> list(indices(3, 2))
+    [(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1)]
+    """
+
+    return(itertools.product(*[range(_) for _ in sizes]))
```

Comment:
no need to add parens around the return:

```python
return itertools.product(*map(range, sizes))
```

In general, I would avoid using `_` like this, most people use that when they want to ignore a variable but must assign it like:

```python
for _ in range(num_retries):
    # code that wants to run `num_retries` times but doesn't
    # need to know the count
```
or
```python
# I just want the first and last, but don't care about the
# middle of an iterator.
first, *_, last = sequence
```

Part of the reason for this convention is that many python shells (default, IPython, bpython, etc) will use `_` to mean: the last evaluated line. For example:
```python
In [16]: 2 + 2
Out[16]: 4

In [17]: _
Out[17]: 4
```
This makes it hard to use a repl to reason about code that uses the name `_` because it will get trampled and reassigned a lot.